### ISSUE TITLE
requireJS can't find crossfilter2

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#1304]
I'm going to file this issue preemptively, because we are changing the way dc.js loads crossfilter and it is going to break requireJS and other AMD package loaders.

https://github.com/crossfilter/crossfilter, the community fork of crossfilter, has a different filename (`crossfilter.js`) from its module name (`crossfilter2`). But requireJS (and presumably other AMD loaders) assumes that the module name will be the same as the filename.

From dc.js 2.0.3 and 2.1.5 onward, `index.js` and the UMD footer of `dc.js` will refer to the module name `crossfilter2` instead of `crossfilter`.

requireJS users will need to rename `crossfilter.js` to `crossfilter2.js` when copying it into position, or use the requireJS paths, or map configuration options to specify that the filename is different from the module name.

http://requirejs.org/docs/api.html#config-paths
http://requirejs.org/docs/api.html#config-map

For further discussion, please see crossfilter/crossfilter#75

**Comment by mtraynham** [comment]
With Webpack, the workaround was:

```
    resolve: {
        alias: {
            crossfilter: 'crossfilter2'
        }
    },
```

Thanks for fixing though, can finally remove that.

**Comment by gordonwoodhull** [comment]
Yes, the problem was that some webpack based projects e.g. angular-cli did not permit changing the config at all. Whereas requireJS is always open for configuration afaik.
